numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -1027,6 +1027,8 @@ typedef void (NpyIter_GetMultiIndexFunc)(NpyIter *iter,
  * eliminate overlap.
  */
 #define NPY_ITER_COPY_IF_OVERLAP            0x00002000
+/* At least one operand uses UPDATEIFCOPY semantics, use context manager */
+#define NPY_ITER_USE_CONTEXT_MANAGER        0x00004000
```

Comment:
Some questions/comments:
  * Just so you know the nditer/NpyIter is mostly a C-Api beast, the python side is not used by far as much.
  * Is the flag really necessary? I guess you would have to check every single operand array, but that does not seem too bad? The flag may not be bad either though, so not sure what I think yet.
  * Why is there an explicit `ResolveUpdateIfCopy` in the `nditer` pywrap code? It seems to me that on the C-Side this should be triggered by the `NpyIter_Deallocate ` call, so it would be more obvious to me if you would just call that. (Hopefully guarding all the other methods against the iterator being `NULL` or so is not too bad, but it should be done anyway).
  * Regarding above point: Right now it seems to me like C-Code using NpyIter is broken in PyPy, because the resolve code is never called.
  * In cpython, if I write `for i in np.nditer(...):` I know the nditer will be deallocated by the time the for loop ends, I could imagine trying to give a warning when this might not work, and we know it from the reference count, since it should be higher if there is any chance of the object living longer. (I don't care whether this warning is a )
  * I suppose PyPy has no way of knowing if a temporary iterator object only exists within the scope of the single for loop? (in which case you could do the same thing as above)